client/network/sync/src/lib.rs

```diff
@@ -2210,10 +2121,16 @@ where
 									.disconnect_peer(id, self.block_announce_protocol_name.clone());
 								continue
 							},
-						};
-
-						if let Some(import) = self.on_block_response(id, req, response) {
-							return Poll::Ready(import)
+							Err(BlockResponseErr::ExtractionFailed(e)) => {
+								debug!(
+									target: "sync",
+									"Failed to extract blocks from peer response {:?}: {:?}.",
+									id,
+									e
+								);
+								self.network_service.report_peer(id, rep::BAD_MESSAGE);
+								continue
```

Comment:
Is it intentional that we don't disconnect a peer in case of failed extraction, like it was with `decode_block_response`?

Reply:
Yes, the change tries to maintain the existing behavior

Before the change:
1. On decode failure: https://github.com/paritytech/substrate/blob/master/client/network/sync/src/lib.rs#L2208, it both disconnects and reports the peer
2. On extraction failure: https://github.com/paritytech/substrate/blob/943c520aa78fcfaf3509790009ad062e8d4c6990/client/network/sync/src/lib.rs#L2075, it only reports the peer, but does not disconnect

`BlockResponseError` was added specifically to distinguish between these two cases